elfi/post_proc.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+from collections import OrderedDict
+import inspect
+
+import elfi.visualization as vis
+
+
+"""
+Implementations for post-processing.
+"""
+
+
+class Result(object):
+    """Container for results from ABC methods. Allows intuitive syntax for plotting etc.
+
+    Parameters
+    ----------
+    samples : OrderedDict of name: np.arrays
+    """
+    def __init__(self, samples, **kwargs):
+        self.samples = samples
+        self.n_samples = len(list(self.samples.values())[0])
+        self.n_params = len(samples)
+
+        # get name of the ABC method
+        stack10 = inspect.stack()[1][0]
+        self.method = stack10.f_locals["self"].__class__.__name__
+
+        # TODO: needed?
+        if any(map(lambda k: k in self.__dir__(), kwargs.keys())):
+            raise KeyError("Conflicting key")
+
+        self.__dict__.update(kwargs)
```

Comment:
Currently the class is a container of samples with added functionality of storing whatever metadata in addition (as arbitrary instance variables). Would be more clear to just have a instance variable `self.metadata = dict()` and use it instead of `self.__dict__`. This also resolves the issue above with checking keys.

I vote for the Result being an abstract interface, from which we inherit actual classes Samples, Likelihood, etc for different interaction functions that work with the concrete results: `class Samples(Result)` For example, if we would have both samples and a likelihood we can evaluate, we would have `class MyResults(Samples, Likelihood)`.